human-dynamics-estimator/src/HumanDynamicsEstimator.cpp

```diff
@@ -68,53 +176,293 @@ bool HumanDynamicsEstimator::configure(yarp::os::ResourceFinder &rf)
+     *  ------Creating a map to fill the y measurements vector with the order used in berdy
      */
-    iDynTree::BerdyHelper berdy;
-    berdy.init(humanModel, humanSensors, berdyOpts);
-    
+    std::vector <iDynTree::BerdySensor> berdySens; 
+    berdySens = m_berdy.getSensorsOrdering();
     
+    std::vector <std::string> vectorID;
+    std::vector <int> vectorOFFSET;
+    for (int index=0; index < berdySens.size(); index ++) 
+    {
+        vectorID[index] = berdySens[index].id;
+        vectorOFFSET[index] = berdySens[index].range.offset;        
+    }
+ 
+    for (int i=0; i<vectorID.size(); i++) 
+    {
+        m_y_map[vectorID[i]] = vectorOFFSET[i];
+    }
+
+    m_y.resize(m_meas);
+    m_y.zero();
+
     /*
-     * ------Open port for sensor data
+     * ------Set priors
      */
-    // TODO: port or device?
+    m_sigmaD_prior.resize(m_eq,m_eq);
+    m_sigmaD_prior.reserve(m_eq);    
+    iDynTree::Triplets Tris_eq;   
+    Tris_eq.addDiagonalMatrix(0,0,1,m_eq);
+    m_sigmaD_prior.setFromTriplets(Tris_eq);
+    Eigen::Map< Eigen::SparseMatrix<double, Eigen::RowMajor> > sigmaD_inv_prior =  toEigen(m_sigmaD_prior);
+    double constSigmaD = rf.find("constSigmaD").asDouble();
+    sigmaD_inv_prior *= (1/constSigmaD);   
     
+    m_sigmad_prior.resize(m_dynV,m_dynV);
+    m_sigmad_prior.reserve(m_dynV); 
+    iDynTree::Triplets Tris_dynV;   
+    Tris_dynV.addDiagonalMatrix(0,0,1,m_dynV);
+    m_sigmad_prior.setFromTriplets(Tris_dynV);
+    Eigen::Map< Eigen::SparseMatrix<double, Eigen::RowMajor> > sigmad_inv_prior =  toEigen(m_sigmad_prior);
+    double constSigmad = rf.find("constSigmad").asDouble();
+    sigmad_inv_prior *= (1/constSigmad); 
+     
+    m_sigmay_prior.resize(m_meas,m_meas);
+    m_sigmay_prior.reserve(m_meas);
+    Eigen::Map< Eigen::SparseMatrix<double, Eigen::RowMajor> > sigmay_inv_prior =  toEigen(m_sigmay_prior);    
     
+    m_mud_prior.resize(m_dynV);
+    m_mud_prior.zero();
+    Eigen::Map< Eigen::VectorXd> mu_d_prior =  toEigen(m_mud_prior);    
+
     /*
-     * ------Wrapper data for a Berdy compatible form --> order of y vector
-     */
-     // TODO: wrapper method
+    * ------Set berdy matrices and gravity
+    */
+    m_berdy.resizeAndZeroBerdyMatrices(m_D, m_bD, m_Y, m_bY);    
+
+    m_gravity.zero(); 
+    m_gravity(2) = -9.81;    
+
+    /* 
+     * ------Creating map for the d dynamics variables vector
+     */  
+    std::vector <iDynTree::BerdyDynamicVariable> berdyDyn; 
+    berdyDyn = m_berdy.getDynamicVariablesOrdering();
+    
+    std::string tempTYPE;
+   
+    for (int index=0; index < berdyDyn.size(); index ++) {   
+        if (berdyDyn[index].type==iDynTree::LINK_BODY_PROPER_ACCELERATION) {
+            tempTYPE = "spatialAcceleration";
+            BerdyOutputRangeMap &map = m_link_map[berdyDyn[index].id];
+            map[tempTYPE] = berdyDyn[index].range;
+        }
+        if (berdyDyn[index].type==iDynTree::NET_INT_AND_EXT_WRENCHES_ON_LINK_WITHOUT_GRAV) {
```

Comment:
These `if` are useless as if the previous one is true the following ones will always be false.
A `case` statement is more appropriate for an enumeration